db/commitlog/commitlog.cc

```diff
@@ -832,54 +859,58 @@ class db::commitlog::segment : public enable_shared_from_this<segment>, public c
 
         // The write will be allowed to start now, but flush (below) must wait for not only this,
         // but all previous write/flush pairs.
-        return _pending_ops.run_with_ordered_post_op(rp, [this, size, off, buf = std::move(buf)]() mutable {
+        co_await _pending_ops.run_with_ordered_post_op(rp, [&]() -> future<> {
```

Comment:
I'm not sure about this lambda coroutine.
It might be safer to move into a method.

Reply:
What are you unsure about? Afaik, lambdas are perfectly valid coroutines, one just needs to declare return type + be very aware of capture mechanics (capture by value into lambda object will _not_ work - but capture reference from surrounding coroutine frame is fine).

Reply:
let's just get this tested thoroughly in debug mode...

Reply:
Since the lambda coroutine's caller is itself a coroutine, and the lambda coroutine lifetime is enclosed by the caller lifetime, this is safe.